app/models/git_repository.rb

```diff
@@ -9,6 +9,11 @@ class GitRepository
     Rails.application.config.samson.cached_repos_dir
   end
 
+  def self.valid_ref_format?(reference)
+    IO.popen(['git', 'check-ref-format', "refs/head/#{reference}"]) { |_| }
```

Comment:
Hmm, okay.  True we might want to rescue the exception.

I also discovered that in production, somehow one of the `Deploy` records in the DB had a commit with a value of `current_account['url'] + '`, which was causing these AirBrake errors.  I think we should also add a validation to objects that expect a commit or a git reference, like `Deploy`, `Release`, or `Job`.